### ISSUE TITLE
loki.source.docker errors for containers attached to multiple networks

### ISSUE THREAD

**Issue by cody42** [grafana/alloy#1014]
### What's wrong?

When scraping logs of docker containers attached to more than one network, there is a constant flow of errors from alloy related to cancelled contexts.

My best guess is that this is related to the deduplication of containers, scraping each only once, but not handling the cancelled contexts for the duplicates gracefully. Logs are scraped successfully though.

A possible workaround is to manually label all but one network with a specific label (e.g. com.example.loki=drop) and use a discovery.relabel component to drop all targets with this label.

### Steps to reproduce

1. Create a container attached to multipe networks:
```yml
services:
  mycontainer:
    image: nginx
    networks:
    - net_a
    - net_b
networks:
  net_a:
    name: net_a
    driver: bridge
  net_b:
    name: net_b
    labels:
      - "com.example.loki=drop"
    driver: bridge
```
```
$ docker compose up -d
```
2. Set up alloy to scrape docker logs.
```
discovery.docker "logs_integrations_docker" {
    host             = "unix:///var/run/docker.sock"
    refresh_interval = "5s"
}
discovery.relabel "logs_integrations_docker" {
    targets = []

    rule {
        target_label = "job"
        replacement  = "integrations/docker"
    }

    rule {
        target_label = "instance"
        replacement  = constants.hostname
    }

    rule {
        source_labels = ["__meta_docker_container_name"]
        regex         = "/(.*)"
        target_label  = "container"
    }

    rule {
        source_labels = ["__meta_docker_container_log_stream"]
        target_label  = "stream"
    }
}
loki.source.docker "logs_integrations_docker" {
    host             = "unix:///var/run/docker.sock"
    targets          = discovery.docker.logs_integrations_docker.targets
    forward_to       = [loki.write.my_loki.receiver]
    relabel_rules    = discovery.relabel.logs_integrations_docker.rules
    refresh_interval = "5s"
}
loki.write "my_loki" {
  endpoint {
    url = ...
  }
}
```

### System information

Linux 5.10 x86_64

### Software version

Grafana Alloy v1.1.1

### Logs

```text
These log lines appear over and over again, more or less every minute:

ts=... level=error msg="error inspecting Docker container" target=cae... component_path=/ component_id=loki.source.docker.logs_integrations_docker id=cae... error="Get \"http://%2Fvar%2Frun%2Fdocker.sock/v1.44/containers/cae.../json\": context canceled"

ts=... level=warn msg="could not transfer logs" component_path=/ component_id=loki.source.docker.logs_integrations_docker target=docker/cae... written=0 container=cae... err="context canceled"

ts=... level=info msg="removing entry from positions file" component_path=/ component_id=loki.source.docker.logs_integrations_docker path=cursor-cae... labels="{..., __meta_docker_container_id=\"cae...\", ..., __meta_docker_network_id=\"f51...\", ...}"

ts=... level=info msg="removing entry from positions file" component_path=/ component_id=loki.source.docker.logs_integrations_docker path=cursor-cae... labels="{..., __meta_docker_container_id=\"cae...\", ..., __meta_docker_network_id=\"a5b...\", ...}"
```

**Comment by cody42** [comment]
Workaround is to use a relabel component inbetween the discovery and the loki source to filter out the duplicate of the container for all but one network:

```hcl
discovery.relabel "logs_integrations_docker_filter" {
         targets = discovery.docker.logs_integrations_docker.targets

         rule {
                action        = "drop"
                source_labels = ["__meta_docker_network_label_com_example_lokii"]
                regex         = "drop"
        }
}

loki.source.docker "logs_integrations_docker" {
        ...
        targets          = discovery.docker.logs_integrations_docker_filter.output
        ...
}
```

Be sure to set the appropriate label `"com.example.loki=drop"` on all but one networks for each container. Depending on the topology, this might become very cumbersome, so this is not a solution in general.
